### ISSUE TITLE
mAP are much lower on UCAS-AOD (= 43.6) and HRSC2016 (= 43.5) with ResNet152

### ISSUE THREAD

**Issue by ri001-lcwang** [Capino512/pytorch-rotation-decoupled-detector#1]
1. GPU: 1*GeForce RTX 2080 Ti.
2. Package version:
    Python 3.7.4
    Numpy 1.18.1
    Pytorch 1.7.1+cu101
    Torchvision 0.8.2+cu101
3. Pretrained weight files:
    wget https://download.pytorch.org/models/resnet101-5d3b4d8f.pth
    wget https://download.pytorch.org/models/resnet152-b121ed2d.pth
4. Test HRSC2016:
    For 1 GPU,
    a. modify train.py
        #batch_size = 12
        #num_workers = 4
        batch_size = 2
        num_workers = 0
        #max_step = 12000
        max_step = 12000*6
        #device_ids = [0, 1]
        device_ids = [0]
        #backbone = resnet.resnet101
        backbone = resnet.resnet152
    b. modify evaluate.py
        #batch_size = 32
        #num_workers = 4
        batch_size = 2
        num_workers = 0
        #backbone = resnet.resnet101
        backbone = resnet.resnet152
5. UCAS-AOD result:
Batch size= 2
Backbone  | Max_step | GPU memory (MB) | Training time (hr) | Plane | Car    | mAP
Resnet101 | 3e4          | 5086                       | 14                         | 44.29 | 42.86 | 43.58
Resnet152 | 3e4          | 6436                       | 16                         | 46.94 | 46.02 | 46.48
Resnet152 | 18e4        | 6436                       | 96                         | 45.68 | 41.51 | 43.6
6. HRSC2016 result:
Batch size= 2
Backbone  |Max_step | mAP
Resnet152 | 7.2e4        | 43.52

7. The paper:
   a. UCAS-AOD: mAP= 97.
   b. HRSC2016: mAP= 94.

Why both mAP in my test are much lower?
Please help me solve this problem.

**Comment by Capino512** [comment]
Thank you for your attention to our research work. Judging from the feedback I have received so far, others have reproduced our results. The following are possible problems for you:
1. Please download the pre-trained weights through the link we gave, not from pytorch (because I redefine the network, the weight keywords have changed). If there is any problem with the download, please contact me again.
2. Too small batch size may result in a decrease in accuracy.

**Comment by ri001-lcwang** [comment]
Thanks for your reply.
I haven't use baiduyun (https://pan.baidu.com) for a long time and my baidu account is locked.
Therefore, I cannot download pretrained weight files from baiduyun.
Please help me.

**Comment by ri001-lcwang** [comment]
> Thank you for your attention to our research work. Judging from the feedback I have received so far, others have reproduced our results. The following are possible problems for you:
> 
> 1. Please download the pre-trained weights through the link we gave, not from pytorch (because I redefine the network, the weight keywords have changed). If there is any problem with the download, please contact me again.
> 2. Too small batch size may result in a decrease in accuracy.

**Comment by Capino512** [comment]
Do you have other cloud disks that you can use, such as Google Drive. I can share the pre-training weight file with you through other network disks.

**Comment by Capino512** [comment]
Or you can convert the downloaded weight file into the required file by modifying the key value, but you should ensure the correctness of the conversion.

**Comment by ri001-lcwang** [comment]
> Do you have other cloud disks that you can use, such as Google Drive. I can share the pre-training weight file with you through other network disks.

Thanks for your reply.
Please share pre-training weight files to the shared folder of my Google drive (15 GB free).
https://drive.google.com/drive/folders/1zk5_NMo8HKl5w6e5gyNAdSxNll6vNrfD?usp=sharing
Thanks.

**Comment by Capino512** [comment]
I don’t have permission to upload files, please make sure it’s turned on.

**Comment by ri001-lcwang** [comment]
> I don’t have permission to upload files, please make sure it’s turned on.

Sorry. I had fixed the permission for uploading and used another google account testing OK.
Thanks.

**Comment by Capino512** [comment]
The pre-training weight files have been uploaded.

**Comment by ri001-lcwang** [comment]
> The pre-training weight files have been uploaded.

Thanks.

Result.
Batch size= 2
1. UCAS-AOD.
Backbone | Max_step | Plane | Car | mAP
Resnet152 | 1.8e5 | 50.08 | 45.26 | 47.67
2. HRSC2016.
Backbone | Max_step | mAP
Resnet152 | 7.2e4 | 44.54

=> mAP are still much lower.
I have uploaded entire code and dataset to https://drive.google.com/drive/folders/1zk5_NMo8HKl5w6e5gyNAdSxNll6vNrfD?usp=sharing
Please help me solve this problem.

**Comment by Capino512** [comment]
I've checked the code you've shared. I will try to help solve your problem. First, I will test if the problem is caused by setting the batch_size too small. I will get back to you if there is any progress.

**Comment by Capino512** [comment]
My local test results did not show any exceptions. Your problem may be caused by an incompatible version of your pytorch (I can't load your saved weight fille via torch.load(path)). The version of pytorch I am using is 1.2.

**Comment by ri001-lcwang** [comment]
> My local test results did not show any exceptions. Your problem may be caused by an incompatible version of your pytorch (I can't load your saved weight fille via torch.load(path)). The version of pytorch I am using is 1.2.

Thanks for you reply.
torchvision >= 0.3 adds many new function and pytorch 1.2 is not compatible with it so I use the following.
torch: 1.7.1+cu101
torchvision: 0.8.2+cu101
Do you mean that mAP of batch_size=12 is the same as batch_size=2 if using pytorch 1.2?

**Comment by Capino512** [comment]
Since there is no permission to update the graphics card driver, we used a lower version of the package (torch=1.2 and torchvision=0.4). I used two strategies for training. One is the same as yours, the other is as follows (recommended), both strategies have achieved accuracy close to batch_size=12.
```
...
    image_size = 768
    lr = 1e-3
    batch_size = 2
    num_workers = 1

    max_step = 12000 * 6
    lr_cfg = [[7500 * 6, lr], [max_step, lr / 10]]
    warm_up = [500 * 6, lr / 50, lr]
    save_interval = 1000 * 6
...
            loss.backward()
            if (current_step + 1) % 6 == 0:
                optimizer.step()
                optimizer.zero_grad()
...
```
